--- rtmpdump-2.3/Makefile.orig	2014-02-02 11:00:29.162772691 +0100
+++ rtmpdump-2.3/Makefile	2014-02-02 11:02:29.859434291 +0100
@@ -2,8 +2,10 @@
 
 prefix=/usr/local
 
+LIBTOOL=libtool -v
 CC=$(CROSS_COMPILE)gcc
 LD=$(CROSS_COMPILE)ld
+INSTALL=install
 
 SYS=posix
 #SYS=mingw
@@ -43,7 +45,7 @@
 THREADLIB=$(THREADLIB_$(SYS))
 SLIBS=$(THREADLIB) $(LIBS)
 
-LIBRTMP=librtmp/librtmp.a
+LIBRTMP=librtmp/librtmp.la
 INCRTMP=librtmp/rtmp_sys.h librtmp/rtmp.h librtmp/log.h librtmp/amf.h
 
 EXT_posix=
@@ -59,11 +61,11 @@
 
 install:	progs
 	-mkdir -p $(BINDIR) $(SBINDIR) $(MANDIR)/man1 $(MANDIR)/man8
-	cp rtmpdump$(EXT) $(BINDIR)
-	cp rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(SBINDIR)
+	@cd librtmp; $(MAKE) install $(MAKEFLAGS)
+	$(LIBTOOL) --tag=CC --mode=install $(INSTALL) rtmpdump$(EXT) $(BINDIR)
+	$(LIBTOOL) --tag=CC --mode=install $(INSTALL) rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(SBINDIR)
 	cp rtmpdump.1 $(MANDIR)/man1
 	cp rtmpgw.8 $(MANDIR)/man8
-	@cd librtmp; $(MAKE) install
 
 clean:
 	rm -f *.o rtmpdump$(EXT) rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT)
@@ -74,17 +76,19 @@
 $(LIBRTMP): FORCE
 	@cd librtmp; $(MAKE) all
 
+LINK=$(LIBTOOL) --tag=CC --mode=link $(CC) $(CFLAGS) $(LDFLAGS)
+
 rtmpdump: rtmpdump.o
-	$(CC) $(LDFLAGS) -o $@$(EXT) $@.o $(LIBS)
+	$(LINK) -o $@$(EXT) $@.o $(LIBS)
 
 rtmpsrv: rtmpsrv.o thread.o
-	$(CC) $(LDFLAGS) -o $@$(EXT) $@.o thread.o $(SLIBS)
+	$(LINK) -o $@$(EXT) $@.o thread.o $(SLIBS)
 
 rtmpsuck: rtmpsuck.o thread.o
-	$(CC) $(LDFLAGS) -o $@$(EXT) $@.o thread.o $(SLIBS)
+	$(LINK) -o $@$(EXT) $@.o thread.o $(SLIBS)
 
 rtmpgw: rtmpgw.o thread.o
-	$(CC) $(LDFLAGS) -o $@$(EXT) $@.o thread.o $(SLIBS)
+	$(LINK) -o $@$(EXT) $@.o thread.o $(SLIBS)
 
 rtmpgw.o: rtmpgw.c $(INCRTMP) Makefile
 rtmpdump.o: rtmpdump.c $(INCRTMP) Makefile
--- rtmpdump-fa8646d/librtmp/Makefile.orig	2015-12-23 20:09:27.000000000 +0100
+++ rtmpdump-fa8646d/librtmp/Makefile	2018-04-14 19:42:42.380311763 +0200
@@ -11,9 +11,11 @@
 LIBDIR=$(DESTDIR)$(libdir)
 MANDIR=$(DESTDIR)$(mandir)
 
+LIBTOOL=libtool -v
 CC=$(CROSS_COMPILE)gcc
 LD=$(CROSS_COMPILE)ld
 AR=$(CROSS_COMPILE)ar
+INSTALL=install
 
 SYS=posix
 CRYPTO=OPENSSL
@@ -78,25 +80,24 @@
 LDFLAGS=$(XLDFLAGS)
 
 
-OBJS=rtmp.o log.o amf.o hashswf.o parseurl.o
+OBJS=rtmp.lo log.lo amf.lo hashswf.lo parseurl.lo
 
-all:	librtmp.a $(SO_LIB)
+all:	librtmp.la
 
 clean:
-	rm -f *.o *.a *.$(SOX) *$(SO_EXT) librtmp.pc
+	$(LIBTOOL) --tag=CC --mode=clean rm
 
-librtmp.a: $(OBJS)
-	$(AR) rs $@ $?
+librtmp.la: $(OBJS)
+	$(LIBTOOL) --tag=CC --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -rpath $(libdir) -o $@ $^ $> $(CRYPTO_LIB)
 
-librtmp$(SO_EXT): $(OBJS)
-	$(CC) $(SO_LDFLAGS) $(LDFLAGS) -o $@ $^ $> $(CRYPTO_LIB)
-	ln -sf $@ librtmp.$(SOX)
-
-log.o: log.c log.h Makefile
-rtmp.o: rtmp.c rtmp.h rtmp_sys.h handshake.h dh.h log.h amf.h Makefile
-amf.o: amf.c amf.h bytes.h log.h Makefile
-hashswf.o: hashswf.c http.h rtmp.h rtmp_sys.h Makefile
-parseurl.o: parseurl.c rtmp.h rtmp_sys.h log.h Makefile
+%.lo:	%.c
+	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(CFLAGS) -c -o $@ $<
+
+log.lo: log.c log.h Makefile
+rtmp.lo: rtmp.c rtmp.h rtmp_sys.h handshake.h dh.h log.h amf.h Makefile
+amf.lo: amf.c amf.h bytes.h log.h Makefile
+hashswf.lo: hashswf.c http.h rtmp.h rtmp_sys.h Makefile
+parseurl.lo: parseurl.c rtmp.h rtmp_sys.h log.h Makefile
 
 librtmp.pc: librtmp.pc.in Makefile
 	sed -e "s;@prefix@;$(prefix);" -e "s;@libdir@;$(libdir);" \
@@ -105,17 +106,10 @@
 		-e "s;@PUBLIC_LIBS@;$(PUBLIC_LIBS);" \
 		-e "s;@PRIVATE_LIBS@;$(PRIVATE_LIBS);" librtmp.pc.in > $@
 
-install:	install_base $(SO_INST)
-
-install_base:	librtmp.a librtmp.pc
+install:	librtmp.la librtmp.pc
 	-mkdir -p $(INCDIR) $(LIBDIR)/pkgconfig $(MANDIR)/man3 $(SODIR)
 	cp amf.h http.h log.h rtmp.h $(INCDIR)
-	cp librtmp.a $(LIBDIR)
+	$(LIBTOOL) --tag=CC --mode=install $(INSTALL) librtmp.la $(LIBDIR)
 	cp librtmp.pc $(LIBDIR)/pkgconfig
 	cp librtmp.3 $(MANDIR)/man3
 
-install_so:	librtmp$(SO_EXT)
-	cp librtmp$(SO_EXT) $(SODIR)
-	$(INSTALL_IMPLIB)
-	cd $(SODIR); ln -sf librtmp$(SO_EXT) librtmp.$(SOX)
-
